### DISCUSSION TITLE
dts-lint error ERR_UNSUPPORTED_ESM_URL_SCHEME

### DISCUSSION THREAD

**Discussion by ts-lover** [DefinitelyTyped/DefinitelyTyped#67600] [_Misc]
Hello, I'm trying to add some types to the DefinitelyTyped repository for an old CommonJS package.

I created a working folder in the DT repo and added the required files using DTS-GEN, and added the declarations to index.d.ts

For testing, I created the folder inside `node_modules/@types` with folder name matching the name of the package and placed copy of the `index.d.ts` file in there.

In the `-tests.ts` file in my woking folder I am importing the package like this:
```
import CjsLib = require('cjs-lib');
CjsLib.method1('test'); // $ExpectType string
```
In this file I can see the correct declarations coming from the `index.d.ts` for the library object and the method. The `tsconfig.json` file generated by DTS-Gen has the module set to `CommonJS`, and everything looks correct.

But when I try to run the test file using `npm run test cjs-lib` it gives an error that says

> Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader.

If I manually compile the `-test.ts` file with tsc, it looks like this:
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CjsLib = require("cjs-lib");
CjsLib.method1('test'); // $ExpectType string
```
Which I think looks correct..?

I'm using current version of nodejs/npm (21.3.0 / 10.2.4) and VSCode on Windows 10

What am I doing wrong or missing?

Edit
I've also installed PNPM and tried running the tests with that but I get the same `ERR_UNSUPPORTED_ESM_URL_SCHEME` error.

I've tried installing and using different Node versions like 14, 16, and 20 and I see the error using any of them.

I also tried rewriting the library the types are for as a global library instead of a commonjs module, and updated the types/tests accordingly. Even like this I see the same error.

**Comment by jakebailey** [comment]
This should now be fixed via https://github.com/microsoft/DefinitelyTyped-tools/pull/855.
